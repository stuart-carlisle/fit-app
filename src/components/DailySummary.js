import React from 'react'
import DailySummaryItem from './DailySummaryItem'
import moment from 'moment'

export const DailySummary = () => {
    const today = moment().startOf('day')
    const yesterday = moment().startOf('day').subtract(1, 'days')
    const twoDaysAgo = moment().startOf('day').subtract(2, 'days')
    const threeDaysAgo = moment().startOf('day').subtract(3, 'days')
    
    return(
        <div>
            <DailySummaryItem key={today.valueOf()} date={today}/>
            <DailySummaryItem key={yesterday.valueOf()} date={yesterday}/>
            <DailySummaryItem key={twoDaysAgo.valueOf()} date={twoDaysAgo}/>
            <DailySummaryItem key={threeDaysAgo.valueOf()} date={threeDaysAgo}/>
        </div>
    )
}

export default DailySummary
