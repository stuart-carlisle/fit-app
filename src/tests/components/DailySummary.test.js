import React from 'react'
import { shallow } from 'enzyme'
import  DailySummary   from '../../components/DailySummary'

test('Should render the daily summary component correctly',()=>{  
    const wrapper = shallow(<DailySummary />)
    expect(wrapper).toMatchSnapshot()
    
})