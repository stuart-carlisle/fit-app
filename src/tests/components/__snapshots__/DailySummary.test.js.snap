// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the daily summary component correctly 1`] = `
<div>
  <Connect(DailySummaryItem)
    date={"1969-12-31T23:00:00.000Z"}
    key="-3600000"
  />
  <Connect(DailySummaryItem)
    date={"1969-12-30T23:00:00.000Z"}
    key="-90000000"
  />
  <Connect(DailySummaryItem)
    date={"1969-12-29T23:00:00.000Z"}
    key="-176400000"
  />
  <Connect(DailySummaryItem)
    date={"1969-12-28T23:00:00.000Z"}
    key="-262800000"
  />
</div>
`;
